### PR TITLE
Reworks scatter.

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Scatter will now always trigger, the value of scatter now being its max deviation(remember, its any place between that angle), gun values were changed(needs testing to see if theyre good) to account for that

## Why It's Good For The Game
Scatter now is more present in the game and can be more used for balance and it ain't RNG anymore

## Changelog
:cl:
balance: scatter now always happen, its value being the max angle that your gun scatter will go(to any side).
balance: due to scatter change most gun scatter value has been changed.
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
keeping this draft so i can track my changes easier

**Comment by SplinterGP** [comment]
needs testing to see gun values change result and to check if shotguns+burst work properly

**Comment by XSlayer300** [comment]
so basically dynamic accuracy?

File: code/modules/projectiles/guns/revolvers.dm [hunk 1]
```diff
@@ -279,15 +279,15 @@
 	starting_attachment_types = list(
 		/obj/item/attachable/mateba_longbarrel,
 	)
-	attachable_offset = list("muzzle_x" = 20, "muzzle_y" = 18,"rail_x" = 16, "rail_y" = 21, "under_x" = 22, "under_y" = 15, "stock_x" = 22, "stock_y" = 15)
+	attachable_offset = list("muzzle_x" = 28, "muzzle_y" = 18,"rail_x" = 16, "rail_y" = 21, "under_x" = 22, "under_y" = 15, "stock_x" = 22, "stock_y" = 15)
```
Comment (SplinterGP) [inline code/modules/projectiles/guns/revolvers.dm:282]:
```suggestion
	attachable_offset = list("muzzle_x" = 20, "muzzle_y" = 18,"rail_x" = 16, "rail_y" = 21, "under_x" = 22, "under_y" = 15, "stock_x" = 22, "stock_y" = 15)
```

File: code/modules/projectiles/guns/revolvers.dm [hunk 2]
```diff
@@ -279,15 +279,15 @@
...
 
 	damage_mult = 0.80
 	damage_falloff_mult = 1.5
 	fire_delay = 0.2 SECONDS
 	aim_fire_delay = 0.3 SECONDS
 	recoil = 0
 	accuracy_mult = 1.1
-	scatter = 10
+	scatter = 6.5
```
Comment (SplinterGP) [inline code/modules/projectiles/guns/revolvers.dm:290]:
```suggestion
	scatter = 10
```

**Review by BraveMole — APPROVED** [review]

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -283,9 +283,9 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	attach_shell_speed_mod = -1
 	accuracy_mod = 0.1
 	recoil_mod = -2
-	scatter_mod = -5
+	scatter_mod = -1
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:286]:
```suggestion
	scatter_mod = -2
```

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -283,9 +283,9 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
...
 	recoil_unwielded_mod = -3
-	scatter_unwielded_mod = -5
+	scatter_unwielded_mod = -1
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:288]:
```suggestion
	scatter_unwielded_mod = -2
```

**Comment by Lumipharon** [comment]
Playing around with this while test merged I'm not sure about this. Due to the super short distances in ss13, small amounts of scatter are effectively meaningless, at least when shooting shit on screen. Other guns basically can't hit anymore however (like T39 one handed, which while questionably effective, did work with RC and gyro).

So really the main change I noticed is some guns where you've reduced the unwielded scatter are now significantly better, as the small amount of scatter they have doesn't really do anything.

**Comment by SplinterGP** [comment]
stop TM'ing it aaa

**Review by Knightfall5 — COMMENTED** [review]
Deployed guns are not receiving the reduced scatter, and are basically unusable. The T-27 for example has 30 scatter in your hands and should be getting -27 for a total of 3 when deployed, but still has 30 scatter.

**Pull request merged**